pyicloud/services/drive.py

```diff
@@ -53,15 +61,23 @@ def get_file(self, file_id, **kwargs):
             params=file_params,
         )
         if not response.ok:
-            return None
-        url = response.json()["data_token"]["url"]
-        return self.session.get(url, params=self.params, **kwargs)
+            self._raise_error(response.status_code, response.reason)
+        package_token = response.json().get("package_token")
+        data_token = response.json().get("data_token")
+        if data_token and data_token.get("url"):
+            return self.session.get(data_token["url"], params=self.params, **kwargs)
+        elif package_token and package_token.get("url"):
+            return self.session.get(package_token["url"], params=self.params, **kwargs)
+        else:
+            raise KeyError("'data_token' nor 'package_token'")
 
     def get_app_data(self):
         """Returns the app library (previously ubiquity)."""
         request = self.session.get(
             self._service_root + "/retrieveAppLibraries", params=self.params
         )
+        if not request.ok:
+            self._raise_error(request.status_code, request.reason)
```

Comment:
Why not make `raise_error` take the request and test `request.ok` ? Name it `raise_if_error`

Reply:
Elegant solution, thanks.